bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/EntryLocationIndexAsync.java

```diff
@@ -0,0 +1,94 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.bookie.storage.ldb;
+
+import com.google.common.collect.Iterables;
+
+import java.io.IOException;
+
+import org.apache.bookkeeper.bookie.EntryLocation;
+import org.apache.bookkeeper.bookie.storage.ldb.KeyValueStorage.Batch;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.stats.StatsLogger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Maintains an index of the entry locations in the EntryLogger.
+ *
+ * <p>Asynchronous write mode class.
+ */
+public class EntryLocationIndexAsync extends EntryLocationIndex {
+
+    private static final Logger log = LoggerFactory.getLogger(EntryLocationIndexAsync.class);
+
+    public EntryLocationIndexAsync(ServerConfiguration conf, KeyValueStorageFactory storageFactory, String basePath,
+                                   StatsLogger stats) throws IOException {
+        super(conf, storageFactory, basePath, stats);
+    }
+
+    @Override
+    public void addLocation(long ledgerId, long entryId, long location) throws IOException {
+        addLocation(null, ledgerId, entryId, location);
+        locationsDb.sync();
+    }
+
+    @Override
+    public Batch newBatch() {
+        return null;
+    }
+
+    @Override
+    public void updateLocations(Iterable<EntryLocation> newLocations) throws IOException {
+        if (log.isDebugEnabled()) {
+            log.debug("Update locations -- {}", Iterables.size(newLocations));
+        }
+
+        // Update all the ledger index pages with the new locations
+        for (EntryLocation e : newLocations) {
+            if (log.isDebugEnabled()) {
+                log.debug("Update location - ledger: {} -- entry: {}", e.ledger, e.entry);
+            }
+            addLocation(null, e.ledger, e.entry, e.location);
+        }
+        locationsDb.sync();
```

Comment:
Here why use locationsDb.sync(); to replace the following logic: 
batch.flush();
batch.close();

Reply:
What is the difference between locationsDb.sync(); and batch.flush();?
In the end it is called db.write(optionSync, emptyBatch);
![image](https://user-images.githubusercontent.com/19296967/156912154-112c2943-72ea-442a-b541-0a41ff15b7df.png)
![image](https://user-images.githubusercontent.com/19296967/156912162-6408a865-9c49-4a5c-a803-f1fda97d1ddc.png)

It doesn't look asynchronous here.

Reply:
sync is different from async,

- sync mode:  
      1.  create a batch; 
      2.  add msg to the batch
      3.  call method(batch.flush) to flush the batch

- sync mode:  
      1. just call method(locationsDb.sync)  to write the data
      2. the rocksdb server will be timed to flush the data